Automation Examples/010.md

```diff
@@ -0,0 +1,200 @@
+
+After pressing Enter add the following workaround.
+
+```G1ANT
+        try errorcall Empty
+            window ‴✱Confirm Save As✱‴
+            keyboard ⋘LEFT⋙⋘ENTER⋙
+        end try
+```
+
+If the window is displayed, click "yes" to save the file. If the window is not displayed, the script would fail so you surround this code with `try` block, add errorcall argument to it so that it runs smoothly without showing any error messages. Add the "Empty" procedure at the end of your script of course.
+
+```G1ANT
+procedure Empty
+end procedure
+```
+
+#### 10. Sometimes you encounter the second problem which is that some company webpages cannot be loaded, the loading wheel is moving and doesn't stop, the `selenium.waitforvalue` command throws timeout exception - fix this issue
+
+So you modify this line:
+
+```G1ANT
+        selenium.waitforvalue ‴return document.getElementsByClassName("date").length > 0‴ expectedvalue true
+```
+
+And add errorjump argument to it:
+
+```G1ANT
+        selenium.waitforvalue ‴return document.getElementsByClassName("date").length > 0‴ expectedvalue true errorjump MoveToNextCompany
+```
+
+And specify the label before `selenium.seturl` command
+
+```G1ANT
+        label MoveToNextCompany
+```
+
+# Final Code
+
+```G1ANT
+addon selenium version 4.100.19170.929
+addon core version 4.100.19170.929
+addon language version 4.100.19163.1112
+
+♥url = http://nseindia.com/corporates/content/eq_research_reports_listed.htm
+♥basePath = C:\Users\dell\Desktop\Automation Anywhere
+
+call DownloadPdfFiles
+
+procedure DownloadPdfFiles
+    selenium.open chrome url ♥url
+    
+    for ♥i from 2 to 198 step 4
+        selenium.waitforvalue ‴return document.getElementsByClassName("normalText").length > 0‴ expectedvalue true
+        selenium.runscript ‴return document.getElementsByClassName("normalText")[♥i].innerText‴
+        ♥companyName = ♥result
+        selenium.runscript ‴document.getElementsByClassName("normalText")[♥i+1].getElementsByTagName("a")[0].click()‴
+        selenium.waitforvalue ‴return document.getElementsByClassName("date").length > 0‴ expectedvalue true errorjump MoveToNextCompany
+        selenium.runscript ‴document.getElementsByClassName("date")[2].getElementsByTagName("a")[0].click()‴
+        ♥dateformat = yyyy-MM-dd
+        ♥path = ♥basePath\♥date\♥companyName
+        keyboard ⋘CTRL+S⋙
+        ⊂
+        System.IO.Directory.CreateDirectory(♥path);
+        ⊃
+        keyboard ⋘CTRL+C⋙
+        ♥clipboard = ♥path\♥clipboard
+        keyboard ⋘CTRL+V⋙
+        keyboard ⋘ENTER⋙
+        
+        try errorcall Empty
+            window ‴✱Confirm Save As✱‴
+            keyboard ⋘LEFT⋙⋘ENTER⋙
+        end try
+        
+        selenium.closetab
+        label MoveToNextCompany
+        selenium.seturl ♥url
+    end for
+    
+    selenium.close
+end procedure
+
+procedure Empty
+end procedure
+```
+
+Of course don't forget to select selenium addon in the Addons panel (View/Addons).
```

Comment:
I would get rid of this "Of course" part.

Reply:
```suggestion
Don't forget to select selenium addon in the Addons panel (View/Addons).
```